lib/Runtime/Types/DictionaryPropertyDescriptor.h

```diff
@@ -6,62 +6,89 @@
+            | (isInitialized ? Flags::IsInitialized : Flags::None) | (isFixed ? Flags::IsFixed : Flags::None) | (usedAsFixed ? Flags::UsedAsFixed : Flags::None)
 #endif
-            PreventFalseReference(true), IsShadowed(false), IsAccessor(true) {}
+            ) { }
 
         DictionaryPropertyDescriptor(TPropertyIndex dataSlot, PropertyAttributes attributes, bool isInitialized = false, bool isFixed = false, bool usedAsFixed = false) :
-            Data(dataSlot), Getter(NoSlots), Setter(NoSlots), Attributes(attributes),
+            Data(dataSlot), Getter(NoSlots), Setter(NoSlots), Attributes(attributes), flags(Flags::PreventFalseReference
 #if ENABLE_FIXED_FIELDS
-            IsInitialized(isInitialized), IsOnlyOneAccessorInitialized(false), IsFixed(isFixed), UsedAsFixed(usedAsFixed),
+            | (isInitialized ? Flags::IsInitialized : Flags::None) | (isFixed ? Flags::IsFixed : Flags::None) | (usedAsFixed ? Flags::UsedAsFixed : Flags::None)
 #endif
-            PreventFalseReference(true), IsShadowed(false), IsAccessor(false) { }
+            ) { }
 
         DictionaryPropertyDescriptor(TPropertyIndex getterSlot, TPropertyIndex setterSlot, PropertyAttributes attributes, bool isInitialized = false, bool isFixed = false, bool usedAsFixed = false) :
-            Data(NoSlots), Getter(getterSlot), Setter(setterSlot), Attributes(attributes),
+            Data(NoSlots), Getter(getterSlot), Setter(setterSlot), Attributes(attributes), flags(Flags::PreventFalseReference | Flags::IsAccessor
 #if ENABLE_FIXED_FIELDS
-            IsInitialized(isInitialized), IsOnlyOneAccessorInitialized(false), IsFixed(isFixed), UsedAsFixed(usedAsFixed),
+            | (isInitialized ? Flags::IsInitialized : Flags::None) | (isFixed ? Flags::IsFixed : Flags::None) | (usedAsFixed ? Flags::UsedAsFixed : Flags::None)
 #endif
-            PreventFalseReference(true), IsShadowed(false), IsAccessor(true) { }
+            ) { }
 
         // this is for initialization.
-        DictionaryPropertyDescriptor() : Data(NoSlots), Getter(NoSlots), Setter(NoSlots), Attributes(PropertyDynamicTypeDefaults),
-#if ENABLE_FIXED_FIELDS
-            IsInitialized(false), IsOnlyOneAccessorInitialized(false), IsFixed(false), UsedAsFixed(false),
-#endif
-            PreventFalseReference(true), IsShadowed(false), IsAccessor(false) { }
+        DictionaryPropertyDescriptor() : Data(NoSlots), Getter(NoSlots), Setter(NoSlots), Attributes(PropertyDynamicTypeDefaults), flags(Flags::PreventFalseReference) { }
 
         template <typename TPropertyIndexFrom>
         void CopyFrom(DictionaryPropertyDescriptor<TPropertyIndexFrom>& descriptor);
 
+    private:
         // SimpleDictionaryPropertyDescriptor is allocated by a dictionary along with the PropertyRecord
-        // so it cannot be allocated as leaf, tag the lower bit to prevent false reference.
-        bool PreventFalseReference:1;
-        bool IsShadowed : 1;
```

Comment:
>IsShadowed [](start = 13, length = 10)

nit: I would have changed these to be methods like `IsShadowed()` and `SetIsShadowed()` and abstract the checking and setting of flags in those methods.